Crash: Deleting a line when there is a diagnostic at end of file

With a file like this....
![image](https://user-images.githubusercontent.com/13532591/70869794-302fdd80-1f41-11ea-8665-b96ea31ce2c5.png)

Pressing `dd` anywhere in the file will crash:
```
[ERROR]    +1ms Global : Exception Invalid_argument("index out of bounds"):
Raised by primitive operation at file "src/Core/Buffer.re" (inlined), line 62, characters 40-58
Called from file "src/Core/Buffer.re", line 94, characters 13-34
Called from file "src/Range.re", line 21, characters 23-34
Called from file "src/Model/Diagnostic.re", line 19, characters 2-42
Called from file "list.ml", line 88, characters 20-23
Called from file "list.ml", line 88, characters 32-39
Called from file "src/Model/Diagnostics.re", line 60, characters 2-51
Called from file "src/Model/Diagnostics.re" (inlined), line 127, characters 46-116
Called from file "src/UI/EditorSurface.re", line 509, characters 17-68
```

Looks like this is an off-by-one issue in the Diagnostics method with the `measure` function it passes to `Range.explode`

--- a/src/Model/Diagnostic.re
+++ b/src/Model/Diagnostic.re
@@ -14,7 +14,10 @@ type t = {
 let create = (~range, ~message, ()) => {range, message};
 
 let explode = (buffer, diagnostic) => {
-  let measure = n => Buffer.getLineLength(buffer, Index.toOneBased(n));
+  let lineCount = Buffer.getNumberOfLines(buffer);
+  let measure = n =>
+    Index.toZeroBased(n) < lineCount
+      ? Buffer.getLineLength(buffer, Index.toZeroBased(n)) : 0;
 
   Range.explode(measure, diagnostic.range)
   |> List.map(range => create(~range, ~message=diagnostic.message, ()));

